plugins/anomaly-detection/anomaly_detection.py

```diff
@@ -383,31 +468,131 @@ def plot_anomaly(spark, init_plot_df, algo_type, algo_func, anomaly_func,
     return ret_plotDF
 
 
-def generate_tad_sql_query(start_time, end_time, ns_ignore_list):
-    sql_query = ("SELECT {} FROM {} "
-                 .format(", ".join(FLOW_TABLE_COLUMNS), table_name))
-    sql_query_extension = []
-    if ns_ignore_list:
-        sql_query_extension.append(
-            "sourcePodNamespace NOT IN ({0}) AND "
-            "destinationPodNamespace NOT IN ({0})".format(
-                ", ".join("'{}'".format(x) for x in ns_ignore_list)))
-    if start_time:
-        sql_query_extension.append(
-            "flowStartSeconds >= '{}'".format(start_time))
-    if end_time:
-        sql_query_extension.append("flowEndSeconds < '{}'".format(end_time))
-    if sql_query_extension:
-        sql_query += "WHERE " + " AND ".join(sql_query_extension) + " "
-    sql_query += "GROUP BY {} ".format(
-        ", ".join(DF_GROUP_COLUMNS + ["flowEndSeconds"]))
+def generate_tad_sql_query(start_time, end_time, ns_ignore_list,
+                           agg_flow=None, pod_label=None, external_ip=None,
+                           svc_port_name=None, pod_name=None):
+    if agg_flow == "pod":
+        if pod_label:
+            inbound_condition = (
+                "ilike(destinationPodLabels, '%{}%') ".format(pod_label))
+            outbound_condition = (
+                "ilike(sourcePodLabels, '%{}%')".format(pod_label))
+        elif pod_name:
+            inbound_condition = (
+                "destinationPodName = '{}'".format(pod_name))
+            outbound_condition = (
+                "sourcePodName = '{}'".format(pod_name))
+        else:
+            inbound_condition = (
+                "destinationPodLabels <> '' ")
+            outbound_condition = (
+                "sourcePodLabels <> ''")
+        if ns_ignore_list:
+            sql_query_extension = (
+                "AND sourcePodNamespace NOT IN ({0}) AND "
+                "destinationPodNamespace NOT IN ({0})".format(
+                    ", ".join("'{}'".format(x) for x in ns_ignore_list)))
+        else:
+            sql_query_extension = ""
+        sql_query = (
+            "SELECT * FROM "
+            "(SELECT {0} FROM {1} WHERE {2} {6} GROUP BY {3}) "
+            "UNION ALL "
+            "(SELECT {4} FROM {1} WHERE {5} {6} GROUP BY {3}) ".format(
+                ", ".join(AGG_FLOW_TABLE_COLUMNS_POD_INBOUND),
+                table_name, inbound_condition, ", ".join(
+                    DF_AGG_GRP_COLUMNS_POD + ['flowEndSeconds']),
```

Comment:
same comment as above

Reply:
Think twice I feel your code is working.
Since you have filtered by pod_name first, even if you are aggregating by pod_labels, it won't change the result...
It may be better to include a comment, just in case someone else is confused like I was.